### DISCUSSION TITLE
部署kruise-daemon对集群ApiServer和ETCD的影响

### DISCUSSION THREAD

**Discussion by steven6718** [openkruise/kruise#1037] [Q&A]
最近希望在项目中使用openKruise的container restart能力和env from metadata 原地升级能力，需要依赖kruise-daemon的部署。看了源码，每个DaemonSet都会list&watch和本node有关的一些资源（虽然已经带了缓存，但是不知道实际压力会增加多少）。

想咨询下，如果集群规模比较大（1000+ node），全量部署kruise-daemon会不会对Apiserver和ETCD造成过大的压力，从而影响集群整体稳定性？
官方有没有一些压测数据，或者建议的最大集群大小，方便我们评估一下是否可以全集群部署，谢谢~

**Comment by FillZpp** [comment]
@steven6718 首先，社区上并没有足够大规模的 Kubernetes 集群用于压测，因为 kruise-daemon 是按节点部署的，没办法用 kubemark 之类的方案来模拟压测。

不过，根据 OpenKruise 在一些企业的实际使用场景来看，从 5k - 1w Node 范围的集群都没有遇到过问题，但如果存在少数超过 1w Node 的集群、并且集群中存在大量持续性的 Pod 更新操作，可能会导致 APIServer 推送 watch 事件的压力变大，但 listwatch 本身对 Etcd 没有影响。

**Reply by steven6718** [reply]
感谢答复，看起来在大规模集群是有实际应用案例的，我们尝试在生产环境使用，做好性能监控~
